src/server/game/Entities/Player/Player.cpp

```diff
@@ -22425,13 +22425,20 @@ bool Player::BuyItemFromVendorSlot(ObjectGuid vendorguid, uint32 vendorslot, uin
     // check current item amount if it limited
     if (crItem->maxcount != 0)
     {
-        if (creature->GetVendorItemCurrentCount(crItem) < pProto->GetBuyCount() * count)
+        if (creature->GetVendorItemCurrentCount(crItem) < count)
         {
             SendBuyError(BUY_ERR_ITEM_ALREADY_SOLD, creature, item, 0);
             return false;
         }
     }
 
+    // Can only buy full stacks
+    if (count % pProto->GetBuyCount())
```

Comment:
This is false.
Take http://www.wowhead.com/item=3371/crystal-vial for example. It is sold in stacks of 20 but you can buy only 1 on retail
Only things bought with special currencies cannot be divided